services/compute/vmSizes.go

```diff
@@ -445,3 +453,26 @@ func GetVirtualMachineSizes() (vmsizes *[]VirtualMachineSizeTypes) {
 	vmsizes = &tmp
 	return
 }
+
+func GetVirtualMachineSizesFromMoc() (vMSizes string, err error) {
```

Comment:
Lets not convert this to string here.
Lets return an array of the new type.

Lets rename the method to GetVirtualMachineSizesDetailed

Reply:
addressed first 2 comments, 3rd comment I took liberty to name it as GetVirtualMachineSizeValues, let me know if you still don't like, I will rename it back